src/state/routing/useRoutingAPITrade.ts

```diff
@@ -59,7 +60,11 @@ export function useRoutingAPITrade<TTradeType extends TradeType>(
   tradeType: TTradeType,
   amountSpecified?: CurrencyAmount<Currency>,
   otherCurrency?: Currency
-): { state: V3TradeState; trade: Trade<Currency, Currency, TTradeType> | null } {
+): {
+  state: V3TradeState
+  trade: Trade<Currency, Currency, TTradeType> | null
+  gasUseEstimateUSD: CurrencyAmount<Token> | null // dollar amount in active chains stabelcoin
```

Comment:
have you considered making that a part of `Trade`? like create a new `Trade` object in the interface that support gas estimates so it's more easily available throughout the app

`class InterfaceTrade extends Trade {
    private gasEstimate: CurrencyAmount<> | null
}
`

Reply:
don't